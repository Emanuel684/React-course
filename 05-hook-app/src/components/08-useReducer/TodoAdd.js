import React from 'react';
import { useForm } from '../../hooks/useForm';


export const TodoAdd = ({ handleAddTodo }) => {

    const [{ todo }, handleInputChange, reset] = useForm({
        todo: ''
    });

    const handleSubmit = (event) => {
        event.preventDefault();

        if (todo.trim().length >= 1) {

            const newTodo = {
                id: new Date().getTime(),
                todo: todo,
                done: false
            };

            handleAddTodo(newTodo);

            reset();
        }
    }

    return (
        <>
            <h4> Agregar TODO </h4>
            <hr />
            <form onSubmit={handleSubmit}>
                <input
                    type="text"
                    name="todo"
                    placeholder="Aprender ..."
                    className="form-control"
                    autoComplete="off"
                    onChange={handleInputChange}
                    value={todo}
                />

                <button
                    type="submit"
                    className="btn btn-outline-primary mt-3 btn-block"
                >
                    Agregar
                </button>
            </form>
        </>
    )
}
