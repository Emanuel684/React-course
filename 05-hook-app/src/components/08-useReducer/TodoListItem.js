import React from 'react'

export const TodoListItem = ({ i, id, todo, done, handleDelete, handleToggle }) => {
    return (
        <li
            key={id}
            className="list-group-item"
            onClick={() => {
                handleToggle(id);
            }}
        > <p
            className={`${done && 'complete'}`}
        > {i + 1}. {todo} </p>
            <button
                className="btn btn-danger"
                onClick={() => {
                    handleDelete(id);
                }}
            >
                Borrar
            </button>
        </li>
    )
}
