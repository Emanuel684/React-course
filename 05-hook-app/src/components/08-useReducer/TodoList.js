import React from 'react';

import { TodoListItem } from './TodoListItem';

export const TodoList = ({ todos = [], handleDelete, handleToggle }) => {
    return (
        <ul className="list-group list-group-flush">
            {
                todos.map((element, i) => (
                    <TodoListItem key={element .id } i={i} id={element.id} todo={element.todo} done={element.done} handleDelete={handleDelete} handleToggle={handleToggle} />
                ))
            }
        </ul>
    )
}
