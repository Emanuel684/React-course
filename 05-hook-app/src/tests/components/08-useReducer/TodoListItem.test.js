import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { TodoListItem } from '../../../components/08-useReducer/TodoListItem';


describe('Pruebas en el componente de <TodoListItem />', () => {
   
    const wrapper = shallow(<TodoListItem /> );

    test('Debe de mostrar correctame', () => {
        
        expect(wrapper).toMatchSnapshot();

    });
    

});