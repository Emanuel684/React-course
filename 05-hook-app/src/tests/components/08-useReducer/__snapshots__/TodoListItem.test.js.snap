// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente de <TodoListItem /> Debe de mostrar correctame 1`] = `
<li
  className="list-group-item"
  onClick={[Function]}
>
   
  <p
    className="undefined"
  >
     
    NaN
    . 
     
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
